sw/device/rom_exts/manifest.md

```diff
@@ -358,3 +385,63 @@ B, without having to re-link or re-compile the image (note we only have one
 entry address). At least initially, we will only create images that can work
 from Slot A, and later we will add support ensuring these images are linked in
 a way that they can run from Slot B without modification.
+
+
+## How To Validate a ROM_EXT Image
+
+Below we provide an abstract definition of how to sign and how to verify a
+ROM_EXT image. This description is provided so that there is a specification of
+what the signing and validation systems are implementing, so that alternate
+implementations may be produced if necessary.
+
+Notation:
+
+*   `rom_ext` - ROM_EXT Image, including manifest and contents (Variable Length)
+    `signed_area(rom_ext)` denotes just the area between the start of the signed
+    contents and the end of the image.
+*   `device_usage_value` - Device-Calculated Usage Constraint Value (fixed
+    length). This is the `device_usage_value` value described above.
+*   `exponent` - RSA Key Exponent
+*   `private_key` - RSA Private Key
+*   `pub_key` - RSA Public Key
+*   `signature` - ROM_EXT RSA Signature,
+*   `||` - Denotes concatenation without a delimiter.
+*   `:=` - Denotes assignment.
+
+### Algorithms:
+
+*   RSA Signing and Verification is done with 3072-bit keys.
+*   The Digest value in the signature is padded using PKCS#1 v1.5.
+*   The SHA2 Operation may be done with SHA2-265, SHA2-384, or SHA2-512,
+    depending on the Signature Algorithm Identifier provided in the manifest.
+*   The combined device usage value and signed area of the ROM_EXT image are
+    padded using PKCS#1 v1.5.
+
+### Signing
+
+This is usually done on a host machine, which does not have access to a real OT
+device. Thus there has to be a way for the host signer to predict the value of
+`device_usage_value`, as it would be calculated on-device.
+
+The `private_key` is never available on-device, in this scheme.
+
+```
+device_usage_value := predict_device_usage_constraint_value(rom_ext, expected_system_state)
+signature := RSASign(Pad_PKCS1_v15(device_usage_value || signed_area(rom_ext)), private_key)
+```
+
+### Validation
+
+The intention here is to reverse the signing process, except the
+`device_usage_value` can usually be calculated "for real", as we're likely to be
+on-device. If we're doing validation off-device, the device_usage_value will
+again have to be predicted, as it is when signing.
+
+This document does not cover how a device should ensure that `public_key` is
+allowed to validate images for a given device.
+
+```
+actual_digest := SHA2(PKCS1_v15(device_usage_value || signed_area(rom_ext)))
+expected_digest := RSAVerify(signature, public_key)
+result := (actual_digest == expected_digest)
```

Comment:
Similar to above:
```
message := device_usage_value || signed_area(rom_ext)
result := RSASSA-PKCS1-V1_5-VERIFY(public_key, message, signature)
```
and add a reference to https://tools.ietf.org/html/rfc3447#section-8.2.2?

Reply:
Similar to above, we need to pad `message` before we verify it.